### PR TITLE
@russell/fix fieldpath

### PR DESCRIPTION
**Pull request by russellwheatley**
bug fix for #3178

### Checklist

- [ ] Supports `Android`
- [ ] Supports `iOS`
- [ ] `e2e` tests added or updated in packages/**/e2e
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by JoaoPauloCMarra** [comment]
👀

File: packages/firestore/lib/FirestoreQueryModifiers.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default class FirestoreQueryModifiers {
   }
 
   get filters() {
-    return this._filters;
+    return this._filters.map(f => ({ ...f, fieldPath: f.fieldPath._toArray() }));
```
Comment (Ehesp) [inline packages/firestore/lib/FirestoreQueryModifiers.js]: Doesn't this just need to be:

`this._filters.map(f => fieldPath._toArray());`

File: packages/firestore/lib/FirestoreQueryModifiers.js [hunk 2]
```diff
@@ -162,7 +162,7 @@ export default class FirestoreQueryModifiers {
 
   where(fieldPath, opStr, value) {
     const filter = {
-      fieldPath: fieldPath._toArray(),
+      fieldPath: fieldPath,
```
Comment (Ehesp) [inline packages/firestore/lib/FirestoreQueryModifiers.js]: Can remove the value

File: packages/firestore/lib/FirestoreQueryModifiers.js (hunk 1 above)
Comment (Ehesp) [reply packages/firestore/lib/FirestoreQueryModifiers.js]: Everything being passed into the `_filters` array should be a `FieldPath`?

Comment (russellwheatley) [reply packages/firestore/lib/FirestoreQueryModifiers.js]: no, the filters are the arguments passed into `where` query. So each filter looks like ->` where(FieldPath, equalityOperator, value)`

Comment (russellwheatley) [reply packages/firestore/lib/FirestoreQueryModifiers.js]: you can actually see this below in the code: `filter.operator`, 'filter.fieldPath'. Lines 191 & 192.

**Pull request merged**